File: HomeServer/dao/switchtypedao.py
from config.databaseconfig import *
from model.switchtype import SwitchType


class SwitchTypeDao(object):

    @db_session
    def create_switch_type(self, name):
        return SwitchType(name=name)

    @db_session
    def read_switch_types(self):
        return SwitchType.select()[:]

    @db_session
    def read_switch_type(self, switch_type_id):
        return SwitchType[switch_type_id]

    @db_session
    def update_switch_type(self, switch_type_id, name):
        st = SwitchType[switch_type_id]
        st.name = name
        return st

    @db_session
    def delete_switch_type(self, switch_type_id):
        SwitchType[switch_type_id].delete()

    @db_session
    def get_switch_type_module_configurations(self, switch_type_id):
        return list(SwitchType[switch_type_id].module_configurations)

File: MobileApplication/app/src/main/java/pl/polsl/mateusz/chudy/mobileapplication/dao/SwitchTypeDao.kt
package pl.polsl.mateusz.chudy.mobileapplication.dao

import android.arch.persistence.room.*
import pl.polsl.mateusz.chudy.mobileapplication.model.SwitchType

/**
 *
 */
@Dao
interface SwitchTypeDao {

    @Query("select * from switchType")
    fun getSwitchTypes(): List<SwitchType>

//    @Query("select * from switch_type where switchType_id = :switchTypeId")
//    fun getSwitchType(switchTypeId: Long): SwitchType
//
//    @Insert
//    fun createSwitchType(switchType: SwitchType)
//
//    @Update
//    fun updateSwitchType(switchType: SwitchType)

//    @Query("delete from switch_type where switchType_id = :switchTypeId")
//    fun deleteSwitchType(switchTypeId: Long)

    @Delete
    fun deleteSwitchType(switchType: SwitchType)
}